### ISSUE TITLE
BatchNorm don't work as expected

### ISSUE THREAD

**Issue by wangjiawen2013** [tracel-ai/burn#2642]
Hi,
Here is the output of pytorch `BatchNorm1d`, it worked well:
![image](https://github.com/user-attachments/assets/52ba0ff6-29d1-499d-bef4-fed01ddb48e7)

and here is the rust burn `BatchNorm`, the output is weired:
![image](https://github.com/user-attachments/assets/cff494d6-7336-43ef-90b0-e0724a0a9b7e)

Which one is correct ? How to make them consistent ?
